src/plugins/timeseries-plugin.ts

```diff
@@ -0,0 +1,233 @@
+        this.interactionsEngine.on('hover', this.handleHover.bind(this));
+
+        this.interactionsEngine.on('up', this.handleMouseUp.bind(this));
+    }
+
+    handlePositionChange(position: { deltaX: number }) {
+        const startPositionX = this.renderEngine.parent.positionX;
+
+        this.interactionsEngine.setCursor('grabbing');
+
+        this.renderEngine.tryToChangePosition(position.deltaX);
+
+        if (startPositionX !== this.renderEngine.parent.positionX) {
+            this.renderEngine.parent.render();
+        }
+    }
+
+    handleMouseUp() {
+        this.interactionsEngine.clearCursor();
+    }
+
+    setData(data: TimeseriesPoint[]) {
+        this.data = data;
+
+        let min = Number.MAX_VALUE;
+        let max = Number.MIN_VALUE;
+        let first = Number.MAX_VALUE;
+        let last = Number.MIN_VALUE;
+
+        this.data.forEach(([ts, v]) => {
+            if (v < min) {
+                min = v;
+            }
+            if (v > max) {
+                max = v;
+            }
+
+            if (ts < first) {
+                first = ts;
+            }
+            if (ts > last) {
+                last = ts;
+            }
+        });
+
+        this.summary = {
+            min,
+            max,
+            first,
+            last,
+        };
+    }
+
+    calcRect(start: number, duration: number, isEnd: boolean) {
+        const w = duration * this.renderEngine.zoom;
+
+        return {
+            x: this.renderEngine.timeToPosition(start),
+            w: isEnd ? (w <= 0.1 ? 0.1 : w >= 3 ? w - 1 : w - w / 3) : w,
+        };
+    }
+
+    handleHover(region: HitRegion<number> | null) {
+        this.hoveredRegion = region;
+    }
+
+    override renderTooltip() {
+        if (this.hoveredRegion) {
+            const round = (v) => Math.round(v * 100) / 100;
+            const data = this.hoveredRegion.data as HitRegionData;
+
+            this.renderEngine.renderTooltipFromData(
+                [
+                    { text: `Value: ${round(data.v)}` },
+                    {
+                        text: `Timestamp: ${round(data.ts)}ms`,
+                    },
+                ],
+                this.interactionsEngine.getGlobalMouse()
+            );
+            return true;
+        }
+        return false;
+    }
+
+    override render() {
+        if (this.data?.length === 0 || !this.summary) {
+            return;
+        }
+
+        const timestampEnd = this.renderEngine.positionX + this.renderEngine.getRealView();
+        const timestampStart = this.renderEngine.positionX;
+
+        const padding = 5;
+        const heightPerValueUnit = (this.height - padding) / Math.max(this.summary.max - this.summary.min, 1);
+
+        const normalizeValue = (v: number) => {
+            return this.height - v * heightPerValueUnit;
+        };
+
+        const d: TimeseriesPointWithIndex[] = [];
+
+        this.data.forEach(([ts, v], idx) => {
```

Comment:
can we use more descriptive names instead of `d`, `v`, `ts` ...
its hard to follow

Reply:
sure - will rename - wasn't sure if an {} is better than a tuple in this case tbh.